storage/coin_storage_test.go

```diff
@@ -373,6 +397,45 @@ func TestCoinStorage(t *testing.T) {
 		assert.Equal(t, &types.CoinIdentifier{Identifier: "coin4"}, coinIdentifier)
 		assert.Equal(t, blockIdentifier, block)
 	})
+
+	t.Run("AddCoins", func(t *testing.T) {
```

Comment:
This test only tries to add coins after we've added blocks. We should also attempt to add coins before we've processed any blocks.